src/robot/client.ts

```diff
@@ -140,20 +143,32 @@ export class RobotClient extends EventDispatcher implements Robot {
       let retries = 0;
       // eslint-disable-next-line no-console
       console.debug('connection closed, will try to reconnect');
-      void backOff(() =>
-        this.connect().then(
-          () => {
-            // eslint-disable-next-line no-console
-            console.debug('reconnected successfully!');
-            events.emit(RECONNECTED, {});
-          },
-          (error) => {
-            // eslint-disable-next-line no-console
-            console.debug(`failed to reconnect - retries count: ${retries}`);
-            retries += 1;
-            throw error;
-          }
-        )
+      void backOff(
+        () =>
+          this.connect().then(
+            () => {
+              // eslint-disable-next-line no-console
+              console.debug('reconnected successfully!');
+              events.emit(RECONNECTED, {});
+            },
+            (error) => {
+              // eslint-disable-next-line no-console
+              console.debug(`failed to reconnect - retries count: ${retries}`);
+              retries += 1;
+              if (retries === this.webrtcOptions?.reconnectMaxAttempts) {
+                console.log(
+                  `reached max attempts: ${this.webrtcOptions.reconnectMaxAttempts}`
+                );
+              }
+              throw error;
+            }
+          ),
+        {
+          // default values taken from `exponential-backoff` library
+          maxDelay:
+            this.webrtcOptions?.reconnectMaxWait || Number.POSITIVE_INFINITY,
+          numOfAttempts: this.webrtcOptions?.reconnectMaxAttempts || 10,
```

Comment:
is there an acceptable range for these numbers? we should probably check that maxDelay and numOfAttempts aren't negative, for example and throw an exception on client startup if configuration is wrong